aio: improve code tabs styling

petebacondarwin opened the pull request:
Closes #23985

petebacondarwin committed:
feat(aio): render hover status on code-tabs

The Material Design spec states that there should be a change
of background when hovering over a tab label.

See https://material.io/design/components/tabs.html#states

Related to #23985

petebacondarwin committed:
feat(aio): display code-tabs in a Material Design "card"

This helps to connect the "content" of the tab to its label.

Closes #23985

mary-poppins commented:
You can preview 5203954 at https://pr24027-5203954.ngbuilds.io/.

gkalpak commented on aio/src/app/custom-elements/code/code-tabs.component.ts:
```diff
@@ -39,6 +40,7 @@ export interface TabInfo {
         </aio-code>
       </mat-tab>
     </mat-tab-group>
```
Indentation please 😇

gkalpak commented on aio/src/styles/2-modules/_code.scss:
```diff
@@ -3,14 +3,13 @@ code-example, code-tabs {
-code-tabs mat-tab-body {
+code-example {
+  border: .5px solid $lightgray;
+  border-radius: 5px;
```
Why also apply these to `.no-box`?

petebacondarwin replied:
not sure ...

petebacondarwin replied:
I'll revert - unless I can come up with a reason

gkalpak commented on aio/src/styles/2-modules/_code.scss:
```diff
@@ -3,14 +3,13 @@ code-example, code-tabs {
 }
 
-code-example,
-code-tabs mat-tab-body {
```
I find it confusing that tabbed code snippets have different styling (e.g. background, color, etc) than non-tabbed snippets. I'm fine dropping the rounded corners (maybe), but the colors... 😕

petebacondarwin replied:
Do you mean like this https://github.com/angular/angular/issues/23984

gkalpak replied:
No, I am talking about the body of the code. Looking at it again, maybe it is just the background:

**Non-tabbed code block (gray background):**
![non-tabbed-code](https://user-images.githubusercontent.com/8604205/40350591-eb0b6392-5db2-11e8-850b-7220f53b5040.png)

**Tabbed code block (white background):**
![tabbed-code](https://user-images.githubusercontent.com/8604205/40350597-eefa448c-5db2-11e8-9b78-76072ff21574.png)

petebacondarwin replied:
Yes the background is different. I tried the grey background but it really looks awkward with the white tabs.

So... this could also be factored into the discussion of #23984 or we could come up with a solution here.

gkalpak replied:
I prefer the gray background, if you ask me.
But I am not blocking this PR over the background :smiley:

petebacondarwin replied:
How about the converse - we ditch the grey background for all the code blocks?

gkalpak replied:
I like it that code blocks have a different background 😞

petebacondarwin replied:
White is a different background:

<img width="859" alt="screen shot 2018-05-22 at 12 48 58" src="https://user-images.githubusercontent.com/15655/40360589-8d706f28-5dbe-11e8-98b0-df6e2c08d5d8.png">

Currently, code blocks have a slightly darker grey background to lighter grey background of the site.

petebacondarwin replied:
If we combined that white background with blue headers for tabs then we would have a consistent look.

gkalpak replied:
Ha ha, but when/if we switch to white background (as discussed in #23983)...
Anyway, this discussion is getting out of hands. Let's merge this as is and iterate on another, focused issue/PR.

petebacondarwin replied:
OK needs a positive review then :-)

gkalpak reviewed (commented):
I was about to say that there is an unnecessary horizontal scrollbar:

![unnecessay-scrollabr](https://user-images.githubusercontent.com/8604205/40308757-e867a568-5d0f-11e8-802b-02cd6cb40259.png)

(But then I realized it was not introduced by this PR 😳)

petebacondarwin committed:
fixup! feat(aio): display code-tabs in a Material Design "card"

petebacondarwin committed:
fixup! feat(aio): display code-tabs in a Material Design "card"

mary-poppins commented:
You can preview 78f5887 at https://pr24027-78f5887.ngbuilds.io/.

petebacondarwin commented:
I fixed those style changes and added the rounded corners back in. See https://pr24027-78f5887.ngbuilds.io/tutorial/toh-pt3#final-code-review

<img width="1035" alt="screen shot 2018-05-21 at 16 05 30" src="https://user-images.githubusercontent.com/15655/40314581-d1b7944e-5d10-11e8-99bb-4bd089348ecc.png">

If this is acceptable (@jenniferfell and @IgorMinar), I suggest we merge this and then work in #23984 to fix the consistency between the different code blocks.

petebacondarwin committed:
fixup! feat(aio): display code-tabs in a Material Design "card"

petebacondarwin committed:
fixup! feat(aio): display code-tabs in a Material Design "card"

jenniferfell commented:
Unrelated: I like the blue box and black font in @gkalpak screen shot above. Might be a nice way to connect the title color, when we get to that consistency PR. 

For **this** PR: Looks good! Thanks for the quick work. Much easier to see the response to mouse action.
